src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -33,22 +34,28 @@ internal class PackageLoadContext
 
         public IVsSolutionManager SolutionManager { get; private set; }
 
+        public readonly NuGetProjectDependencyVersionLookup DependencyVersionLookup;
+
         public PackageLoadContext(
             IEnumerable<SourceRepository> sourceRepositories,
             bool isSolution,
-            INuGetUIContext uiContext)
+            INuGetUIContext uiContext,
+            NuGetProjectDependencyVersionLookup dependencyVersionLookup)
         {
             SourceRepositories = sourceRepositories;
             IsSolution = isSolution;
             PackageManager = uiContext.PackageManager;
             Projects = (uiContext.Projects ?? Enumerable.Empty<NuGetProject>()).ToArray();
             PackageManagerProviders = uiContext.PackageManagerProviders;
             SolutionManager = uiContext.SolutionManager;
+            DependencyVersionLookup = dependencyVersionLookup;
 
-            _installedPackagesTask = PackageCollection.FromProjectsAsync(Projects, CancellationToken.None);
+            _installedPackagesTask = PackageCollection.FromProjectsAsync(Projects, null, CancellationToken.None);
+            _resolvedPackagesTask = PackageCollection.FromProjectsAsync(Projects, DependencyVersionLookup, CancellationToken.None);
```

Comment:
why do you really need two different package collections? I would expected it to be single resolved packages